tests/lib/rules/no-undefined.js

```diff
@@ -18,7 +18,12 @@ const rule = require("../../../lib/rules/no-undefined"),
 
 const errors = [{ message: "Unexpected use of undefined.", type: "Identifier" }];
 
-const ruleTester = new RuleTester();
+const ruleTester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: 6,
+        sourceType: "module"
```

Comment:
`sourceType: "module"` creates extra scope for modules, so I think it should not apply to all test cases.